src/App.js

```diff
@@ -1,13 +1,42 @@
 import React, { Component } from 'react'
+import changeColor from 'gen-random-color'
 import './App.css'
 
 
 class App extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      color: '#000000',
+      usedColors: []
+    }
+  }
+
+  repeatAction(func, num, delay) {
+    if (!num) 
+      return
+
+    const { usedColors } = this.state
+    const newColor = func(usedColors)
+    
+    this.setState({usedColors: [...usedColors, newColor], color: newColor})
+    
+    setTimeout(() => {this.repeatAction(func, num - 1, delay)}, delay)
+  }
+
+  componentWillMount() {
+    this.repeatAction(changeColor, 10, 1000)
```

Comment:
To me, it seems that the time consuming part of changeColor is that it needs to check against an ever expanding array of colors. The ideal would be to simply increment the hex value by 1 (or use some other type of smart math), so then you'd never need to check the new color against the array, but then you no longer have a truly random color. 

Of course, because you're asking the question there's definitely an answer, but unfortunately I don't know what it is. That said, I would love to learn so please tell me even if this one knocks me out for the job.